template-parts/calendar/calendar.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Calendar partial for the calendar template.
+ *
+ * @package Responsive_Framework
+ */
+
+global $buCalendar;
+
+// Displays the h1 page title and content.
+if ( have_posts() ) {
+	while ( have_posts() ) :
+		the_post();
+		the_title( '<h1 ' . r_page_title_class() . '>', '</h1>' );
+		the_content( '<p class="serif">Read the rest of this page &raquo;</p>' );
+		wp_link_pages( array(
+			'before'         => '<p><strong>Pages:</strong> ',
+			'after'          => '</p>',
+			'next_or_number' => 'number',
+		) );
+	endwhile;
+}
+
+// Displays calendar component.
+$calendar_id  = responsive_calendar_get_calendar_id();
+$calendar_url = responsive_calendar_get_calendar_url();
+
+// If no calendar ID exists, display error message.
+if ( ! $calendar_id ) {
+	?>
+		<p>This site does not have any calendar associated with it.</p>
+	<?php
+} else {
+
+	// Get initial values.
+	$yyymmdd   = responsive_calendar_get_yyyymmdd();
+	$timestamp = responsive_calendar_get_timestamp();
+
+	/*
+	* Check that date falls between:
+	* The year 2000 (http://www.nbc.com/nbc/Late_Night_with_Conan_OBrien/intheyear2000/)
+	* Ten years in the future from the current date
+	*/
+	$boundary_past = strtotime( '2000-01-01 00:00:00', 0 );
+	$boundary_future = strtotime( '+10 years', $now );
```

Comment:
Should these two variables call the responsi boundary_past/future functions above?

Reply:
Yea even though they're only called once it's a step toward moving to less-logic in the templates. It really doesn't matter though, I didn't get as far refactoring at this point in the template as I had hoped. Updated to use the helper functions. 👍